packages/atclient/src/notification.c

```diff
@@ -0,0 +1,306 @@
+#include "atclient/notification.h"
+#include "atclient/connection.h"
+#include "atclient/constants.h"
+#include "cJSON/cJSON.h"
+#include <atchops/uuid.h>
+#include <atlogger/atlogger.h>
+#include <string.h>
+#include <sys/time.h>
+
+void atclient_notify_params_init(atclient_notify_params *params) {
+  params->message_type = NMT_key;
+  params->priority = NP_low;
+  params->strategy = NS_all;
+  params->latest_n = 1;
+  params->notifier = ATCLIENT_DEFAULT_NOTIFIER;
+  params->notification_expiry = 24 * 60 * 60 * 1000; // 24 hours in milliseconds
+}
```

Comment:
Did you forget to do `atclient_atkey_init(&(params->key));` here?

Reply:
I will do another PR for notification.c, please focus on the atkey review